src/js/node/events.js

```diff
@@ -356,52 +357,67 @@ function once(emitter, type, options) {
   });
 }
 
-async function* on(emitter, event, options = {}) {
+function on(emitter, event, options = {}) {
   const signal = options.signal;
-  if (signal?.aborted) throw new AbortError(undefined, { cause: signal?.reason });
```

Comment:
Removed since this part is only executed on creation of the Iterator and not during iteration itself. That part should already be handled by the `signal?.addEventListener("abort", ...)` part later.